palantirnet/federated-search-demo
# 4 x

## Ticket(s)

- See ticket https://www.drupal.org/project/search_api_federated_solr/issues/3191257


## Description

- Creates a 4.x release of Search API Federated Solr for use with Search API Solr 4.x and Solr 7.

## Testing instructions

- Checkout this branch
- If you have one, delete the `src` directory from the project root.
- Run `vagrant up` from the project directory -- this should re-install all project dependencies.
- `vagrant ssh`
- `phing install-all` and check errors.

## Remaining Tasks

We get the following error on `phing install-d8` which suggests a schema mismatch.

```
 [success] Successfully enabled: federated_default_content
 [error]  Drupal\search_api_solr\SearchApiSolrException while indexing item entity:node/2:en: Solr endpoint http://localhost:8983/ bad request (400). ERROR: [doc=9g44bj-federated_search-entity:node/2:en] multiple values encountered for non multiValued field ss_federated_terms: [Age>Adult, Traits>Loyal] in Drupal\search_api_solr\SolrConnector\SolrConnectorPluginBase->handleHttpException() (line 1016 of /var/www/federated-search-demo.local/web/d8/docroot/modules/contrib/search_api_solr/src/SolrConnector/SolrConnectorPluginBase.php). 
```
